homeassistant/components/octoprint/sensor.py

```diff
@@ -72,12 +69,27 @@ def __init__(
         coordinator: DataUpdateCoordinator,
         sensor_name: str,
         sensor_type: str,
+        device_id: str,
     ):
         """Initialize a new OctoPrint sensor."""
         super().__init__(coordinator)
         self._state = None
         self._sensor_name = sensor_name
         self._name = f"{sensor_name} {sensor_type}"
+        self._device_id = device_id
+
+    @property
+    def device_info(self):
+        """Device info."""
+        return {
+            "identifiers": {(COMPONENT_DOMAIN, self._device_id)},
+            "name": self._sensor_name,
+        }
+
+    @property
+    def unique_id(self):
+        """Return a unique id."""
+        return f"{self._name}-{self._device_id}"
```

Comment:
The configured name should not be included in the unique id. It has to both be unique and stable.

Reply:
Swapped to be the sensor name